active/InformationBar/InformationBar.md

```diff
@@ -0,0 +1,622 @@
+
+
+## Message 
+
+Text in your notification will not be a constant length in all languages. For the Title and Message property 
+this may impact whether your notification will expand to a second line. 
+We recommend you avoid positioning based on message length or other UI elements set to a specific language.
+
+The notification will follow standard mirroring behavior when localized to/from languages that are right to left (RTL) or left to right (LTR). The icon will only mirror if there is directionality.
+
+Please view the guidance for 
+[Adjust layout and fonts, and support RTL](https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/adjust-layout-and-fonts--and-support-rtl) 
+for more information about text localization in your notification.
+
+
+# API Notes
+
+### Notable Properties  
+| Name | Description |
+|:-:|:--|
+| IsOpen| Gets or sets a value that determines the visibility of the InfoBar. By default, is set to false. |
+| Severity | Gets or sets a value that indicates the  color and icon to style the InfoBar. It will also define the assistive technology settings. |
+| IsUserDismissable| Gets or sets a boolean that indicates whether the user will be able to dismiss the InfoBar.
+
+
+### Events  
+| Name | Description |
+|:-:|:--|
+| CloseButtonClick | Occurs after the close button has been clicked. |
+| Closed | Occurs after the info bar is closed. |
+| Closing |Occurs just before the info bar begins to close. |
+
+
+# API Details
+
+```c++
+enum InfoBarCloseReason
+{
+    CloseButton = 0,
+    Programmatic = 1,
+};
+
+enum InfoBarSeverity
+{
+    Error = 0,
+    Warning = 1,
+    Success = 2,
+    Default = 3,
+}
+
+runtimeclass InfoBarClosedEventArgs
+{
+    InfoBarCloseReason Reason{ get; };
+};
+
+runtimeclass InfoBarClosingEventArgs
+{
+    InfoBarCloseReason Reason{ get; };
+    Boolean Cancel;
+    Windows.Foundation.Deferral GetDeferral();
+};
+
+unsealed runtimeclass InfoBarTemplateSettings : Windows.UI.Xaml.DependencyObject
+{
+    InfoBarTemplateSettings();
+    Windows.UI.Xaml.Controls.IconElement IconElement;
+
+    static Windows.UI.Xaml.DependencyProperty IconElementProperty{ get; };
+}
+
+unsealed runtimeclass InfoBar : Windows.UI.Xaml.Controls.ContentControl
+{
+    InfoBar();
+
+    String Title;
+    String Message;
+
+    Boolean IsOpen;
+    Boolean IsUserDismissable;
+    Boolean IsIconVisible;
+
+    Windows.UI.Xaml.Controls.Primitives.ButtonBase ActionButton;
+
+    Windows.UI.Xaml.Style CloseButtonStyle;
+    Windows.UI.Xaml.Input.ICommand CloseButtonCommand;
+    Object CloseButtonCommandParameter;
+
+    InfoBarSeverity Severity;
+    IconSource IconSource;
+
+    InfoBarTemplateSettings TemplateSettings{ get; };
+    Object Content{ get; set; };
+    Windows.UI.Xaml.DataTemplate ContentTemplate{ get; set; };
+
+    event Windows.Foundation.TypedEventHandler<InfoBar, Object> CloseButtonClick;
+    event Windows.Foundation.TypedEventHandler<InfoBar, InfoBarClosingEventArgs> Closing;
```

Comment:
The fact that you can trigger `Closing` by setting `IsOpen=false` means that inside the `Closing` handler, if you read the `IsOpen` property, you get `false` which is weird because you expect `Closing` to be raised *before* it closes. (As noted later, if you then cancel the Closing event, this sets `IsOpen` back to `true`.)

If the InfoBar is interactively closed, does `IsOpen` read as true or false during `Closing`?

`NavigationView` has the same problem, since you can set `IsPaneOpen = !IsPaneOpen` and the `-ing` events will be raised even though the transition has already occurred as far as the property is concerned. Are we at least consistent with NavigationView's behavior here?

Reply:
+1 to consistency, see also [FlyoutBase.IsOpen](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutBase.IsOpen)

Reply:
For NavigationView, we're toggling the IsPaneOpen property as soon as the action begins. So in both PaneClosing and PaneClosed events, we report IsPaneOpen as False; in both PaneOpening and PaneOpen we report IsPaneOpen as True.

Reply:
Also, unless I'm doing something wrong, cancelling the event using NavigationViewPaneClosingEventArgs does nothing. The behavior of this
```c#
        private void NavView_PaneClosing(Microsoft.UI.Xaml.Controls.NavigationView sender, Microsoft.UI.Xaml.Controls.NavigationViewPaneClosingEventArgs args)
        {
            bool open = sender.IsPaneOpen;
            args.Cancel = true;
            open = sender.IsPaneOpen;
        }
```
is identical to the behavior of this
```c#
        private void NavView_PaneClosing(Microsoft.UI.Xaml.Controls.NavigationView sender, Microsoft.UI.Xaml.Controls.NavigationViewPaneClosingEventArgs args)
        {
            bool open = sender.IsPaneOpen;
        }
```

Reply:
@YuliKI We have the open issue https://github.com/microsoft/microsoft-ui-xaml/issues/3352 which I plan to look into soon and where I could take a look at IsPaneOpen not being updated correctly when cancelling the close.

Reply:
Thank you, @Felix-Dev, that would be great.

Reply:
Wrapping up this comment thread, yes, the behavior for InfoBar is the same as NavView. Once closing or opening starts, the isOpen property is set to false or true respectively.

Reply:
Thanks @YuliKl and @Felix-Dev for hopping in here. InfoBar and NavigationView _do_ follow the same pattern here that immediately when the action begins IsOpen is set to the intended value. See RaiseClosingEvent and RaiseClosedEvent in [InfoBar.cpp](https://github.com/microsoft/microsoft-ui-xaml/pull/3325/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.cs&file-filters%5B%5D=.h&file-filters%5B%5D=.projitems&file-filters%5B%5D=.shproj&file-filters%5B%5D=.targets&file-filters%5B%5D=.vcxproj&file-filters%5B%5D=.xaml#diff-624e3a295bc0485e93ffbc0dc6e381a4240ab7fc621f7600f3ad401ef3942963)

Reply:
@teaP ping to ensure I'm correct here 😊